deploy/helm/elastic-agent/templates/integrations/_auto_ops/_auto_ops.tpl

```diff
@@ -0,0 +1,125 @@
+{{- define "elasticagent.autoops.init" -}}
+{{- if eq $.Values.autoOps.enabled true -}}
+{{- $presetVal := $.Values.agent.presets.autoOps -}}
+{{- $autoOpsConfig := ((include "elasticagent.autoops.config" $) | fromYaml) -}}
+{{- include "elasticagent.preset.mutate.otelConfig" (list $ $presetVal $autoOpsConfig) -}}
+{{- $autoOpsEnvVars := ((include "elasticagent.autoops.envVars" $) | fromYaml) -}}
+{{- include "elasticagent.preset.mutate.envs" (list $presetVal $autoOpsEnvVars)}}
+{{- end -}}
+{{- end -}}
+
+{{- define "elasticagent.autoops.config" -}}
+receivers:
+  metricbeatreceiver:
+    metricbeat:
+      modules:
+        # Metrics
+        - module: autoops_es
+          hosts: ${env:AUTOOPS_ES_URL}
+          period: 10s
+          metricsets:
+            - cat_shards
+            - cluster_health
+            - cluster_settings
+            - license
+            - node_stats
+            - tasks_management
+        # Templates
+        - module: autoops_es
+          hosts: ${env:AUTOOPS_ES_URL}
+          period: 24h
+          metricsets:
+            - cat_template
+            - component_template
+            - index_template
+    processors:
+      - add_fields:
+          target: autoops_es
+          fields:
+            temp_resource_id: ${env:AUTOOPS_TEMP_RESOURCE_ID}
+            token: ${env:AUTOOPS_TOKEN}
+    output:
+      otelconsumer:
+    telemetry_types: ["logs"]
+
+exporters:
+  otlphttp:
+    headers:
+      Authorization: "AutoOpsToken ${env:AUTOOPS_TOKEN}"
+    endpoint: ${env:AUTOOPS_OTEL_URL}
+
+service:
+  pipelines:
+    logs:
+      receivers: [metricbeatreceiver]
+      exporters: [otlphttp]
+  telemetry:
+    logs:
+      encoding: json
+{{- end -}}
+
+{{- define "elasticagent.autoops.envVars" -}}
+{{- $presetName := "autoOps" -}}
+{{- $agentName := include "elasticagent.preset.fullname" (list $ $presetName) -}}
+extraEnvs:
+  # Always present
+  - name: AUTOOPS_TOKEN
+    valueFrom:
+      secretKeyRef:
+        name: {{ $agentName }}-autoops
+        key: autoops-token
+
+  - name: AUTOOPS_TEMP_RESOURCE_ID
+    valueFrom:
+      secretKeyRef:
+        name: {{ $agentName }}-autoops
+        key: temp-resource-id
+
+  - name: AUTOOPS_OTEL_URL
+    valueFrom:
+      secretKeyRef:
+        name: {{ $agentName }}-autoops
+        key: otel-url
+
+  - name: AUTOOPS_ES_URL
+    valueFrom:
+      secretKeyRef:
+        name: {{ $agentName }}-autoops
+        key: autoops-es-url
+
+  # Only if API key provided
+  {{- if $.Values.autoOps.es_api_key }}
+  - name: ELASTICSEARCH_READ_API_KEY
```

Comment:
please prefix all specific to AutoOps env vars with AUTOOPS, this way we make sure this doesn't collide with any other integration that might get added in the chart in the future

Reply:
The issue is that ELASTICSEARCH_READ_PASSWORD, ELASTICSEARCH_READ_USERNAME, and ELASTICSEARCH_READ_API_KEY are already predefined environment variables. From what I understand, the agent code itself handles all of the authentication configuration, which means we don’t have the flexibility to rename these variables here. I’m not entirely sure why it was designed this way (I believe @pickypg originally set this up), but it seems intentional.

@pickypg, could you help us understand the reasoning behind this approach?

Reply:
Helm / K8s (and other supported mechanisms, like Docker) will pass in those fields and the _code_ picks up their values. That's why you don't see it used in the config itself because of the OTel mode bug (https://github.com/elastic/beats/issues/45102), they cannot be specified if they may be end up unused (and since we support two different forms of auth, inherently one or the other won't be specified).

https://github.com/elastic/beats/blob/a683d8bdfb09f3fb9d0b010eb09b077fcf377b2f/metricbeat/module/elasticsearch/metricset.go#L37-L38

Reply:
Added it under values.schema.json